docs/local_resource.md

```diff
@@ -0,0 +1,100 @@
+---
+title: Run Local and/or Occasional Workflows with Local Resource
+layout: docs
+---
+A Local Resource works like any other resource in your sidebar (which are most
+often some combination of Docker Build instructions and K8s Yaml)--encompasses
+a unit of work, and execute in response to file changes, or
+[manually](https://docs.tilt.dev/manual_update_control.html). For your resource
+`MyGreatService`, when one of its file dependencies changes, the work it executes
+is to build a Docker image and deploy some k8s yaml; for a Local Resource, it's
+to execute an arbitrary command on your local filesystem.
+
+# Run commands locally on file change
+
+Tilt is designed to make it easy to build images, and to sync files/run commands
+in the cloud. For some kinds of development, however, you might want to harness
+Tilt's responsiveness to file changes, but run commands locally instead. These
+cases often fall into the general categories of:
+- *file changes or other artifacts that you want on your local machine (e.g. for
+git commit)*. For instance: when `package.json` changes, run `yarn install`
+(because you want the resulting `yarn.lock` file to exist on your local computer
+so that you can commit it to git)
+
+```python
+local_resource('yarn', cmd='yarn install', deps=['package.json'])
+```
+- *tooling that you have locally but don't want to put on your container images*.
+For instance: rather than pulling/pushing big container images with the Go compiler,
+you'd rather compile your binary locally, and then use Live Update to quickly sync
+the new binary up to a running container in your cluster
+
+See also: [`local_resource` API spec](api.html#api.local_resource).
+
+(Of course, the command you run doesn't have to be on that (only) affects your
+local filesystem. It might, for instance, be a script that is _invoked locally_
+but runs against your k8s cluster.)
+
+## Local Resource + Live Update = <3
+
+Local Resource and Live Update are two great tastes that taste great together.
+A really useful pattern is to have a Local Resource which creates an artifact
+that is then picked up by a Live Update and synced to a running container.
+For example, here's how you might use this combination to generate protobufs:
+```python
+# whenever `helloworld.proto` changes, generate Python protobufs
+local_resource('proto',
+    cmd='protoc --proto_path=src --python_out=build/gen',
+    deps=['helloworld.proto']
+)
+
+# define your hello-world resource: k8s yaml + a Docker build
+k8s_yaml('hello-world.yaml')
+docker_build('hello-world', '.',
+    ignore='helloworld.proto',  # don't step on Local Resource's toes
+    live_update=[ sync('.', '/app') ]
+)
+```
+
+Note that the `docker_build` call specifies `ignore='helloworld.proto'`. This is
+because we DON'T want an edit to that proto file to kick off a Docker build.
```

Comment:
```suggestion
because we DON'T want an edit to that proto file to directly kick off a hello-world build.
```

1. we do want it to kick off a build, transitively
2. the "docker build" vs "live update" bit is a bit ambiguous (is a live update a "docker build"? if we say yes, then we definitely want the "directly" if we say no, then the `ignore` is irrelevant anyway because changing the proto would only ever kick off a live update, not a docker build - also saying no is weird when it is kicking off the live update of a docker build. this is probably just more unexplored ambiguous terminology territory)

Reply:
better?